src/test/run-pass/yk_swt/collect-trace.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 King's College London.
+// Created by the Software Development Team <http://soft-dev.org/>.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(yk_swt)]
+#![feature(libc)]
+
+extern crate core;
+extern crate libc;
+
+use core::yk_swt::{start_tracing, stop_tracing};
+
+pub fn main() {
+    start_tracing();
+    let _ = work();
+    let trace = stop_tracing();
+
+    let len = trace.len();
+    // The default capacity of the trace buffer is 1024. We want to be sure we've tested the case
+    // where it had to be reallocated beyond its starting capacity.
+    assert!(len > 1024);
+
+    for idx in 0..len {
+        let loc = trace.loc(idx).unwrap();
+        assert_ne!(loc.crate_hash, 0); // A crate hash is so unlikely to be zero.
```

Comment:
"So unlikely"?!

Reply:
The wording, definitely bothers me ;) Given that the assertion is incorrect in some cases, I'm also unsure if it's a wise thing to be asserting.

Reply:
OK, let's kill the assert.